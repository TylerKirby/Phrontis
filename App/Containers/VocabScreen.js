import React from 'react';
import { View, Text, Picker } from 'react-native';
import RoundedButton from '../Components/RoundedButton';
import vocab from '../../GreekAppData/vocab.json';

// Styles
import styles from './Styles/VocabScreenStyles'


export default class VocabScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      flip: true,
      deckPosition: 0
    };
    this.flipCard = this.flipCard.bind(this);
    this.correct = this.correct.bind(this);
    this.wrong = this.wrong.bind(this);
    this.showCard = this.showCard.bind(this);
  }

  // Get unit selected from UnitSelectScreen
  componentWillMount() {
    this.setState({
      deck: vocab[this.props.navigation.state.params.selectedChapter]
    })
  }

  flipCard() {
    this.setState({
      flip: !this.state.flip,
    });
  }

  correct() {
    const { deckPosition, deck, flip } = this.state;
    const reset = deckPosition + 1 < deck.length;

    this.setState({
      flip: !flip,
      deckPosition: reset ? deckPosition + 1 : 0,
      deck: deck.splice(deckPosition),
    });

  }

  wrong() {
    const { deckPosition, deck, flip } = this.state;
    const reset = deckPosition + 1 < deck.length;

    this.setState({
      flip: !flip,
      deckPosition: reset ? deckPosition + 1 : 0
    });
  }

  showCard() {
    const { flip, deck, deckPosition } = this.state;
    const lastCard = deck.length < 1
    if (!lastCard) {
      return deck[deckPosition][flip ? 'greek' : 'english']
    }
    return "Great job!"
  }


  render() {

    // Debugging statements:
    console.log('Deck: ', this.state.deck)
    console.log('Flip state: ', this.state.flip)
    console.log('Deck Position', this.state.deckPosition)

    const { flip, deck, deckPosition } = this.state;

    return(
      <View style={styles.mainContainer}>
        <View style={styles.cardContainer}>
          <Text style={styles.cardText}>{this.showCard()}</Text>
        </View>
        {flip ?
        <View style={styles.buttonContainer} >
          <RoundedButton 
            text="flip"
            onPress={this.flipCard}
          />
        </View>
        :
        <View style={styles.buttonContainer}>
          <RoundedButton 
            text="✔️" 
            onPress={this.correct}
          />
          <RoundedButton 
            text="❌"
            onPress={this.wrong}
          />
        </View>
        }
      </View>
    )
  }
}
