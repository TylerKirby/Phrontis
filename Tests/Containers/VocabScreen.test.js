import 'react-native';
import React from 'react';
import VocabScreen from '../../App/Containers/VocabScreen';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

test('VocabScreen component renders correctly', () => {
  const tree = renderer.create(
    <VocabScreen
      navigation={{ state: { params: { selectedChapter: 'unit1' } } }}
    />).toJSON();
  expect(tree).toMatchSnapshot();
});
