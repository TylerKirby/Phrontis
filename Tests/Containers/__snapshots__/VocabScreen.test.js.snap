// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VocabScreen component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0496FF",
      "flex": 1,
      "justifyContent": "center",
      "paddingBottom": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "height": 333.5,
        "justifyContent": "center",
        "width": 562.5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      ἀγορά, ἀγορᾶς, ἡ
    </Text>
  </View>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#D81159",
            "borderRadius": 5,
            "height": 45,
            "justifyContent": "center",
            "marginHorizontal": 25,
            "marginVertical": 10,
            "opacity": 1,
            "width": 150,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginVertical": 10,
              "textAlign": "center",
            }
          }
        >
          FLIP
        </Text>
      </View>
    </View>
  </View>
</View>
`;
